wall_e/src/resources/cogs/ban.py

```diff
@@ -0,0 +1,344 @@
+        e_obj.set_footer(text="Intercepted Moderator Action")
+
+        if e_obj:
+            e_obj.timestamp = date.astimezone(pytz.timezone('Canada/Pacific'))
+            await self.mod_channel.send(embed=e_obj)
+        logger.info(f"[Ban ban()] Message sent to mod channel,{self.mod_channel}, of the ban for {username}.")
+
+    @commands.command()
+    async def initban(self, ctx):
+        """Reads in all guild bans into this ban system"""
+
+        logger.info(f"[Ban initban()] Initban command detected from {ctx.author}")
+
+        try:
+            # audit logs contains info about user who did the banning, the timestamp of the ban, and the reason
+            # however audit logs only go back 3 months, so have to read older bans from the bans list
+            audit_logs = await self.bot.guilds[0].audit_logs(action=discord.AuditLogAction.ban).flatten()
+            bans = await self.bot.guilds[0].bans()
+        except Exception as e:
+            logger.info(f'[Ban initban()] error while fetching ban data: {e}')
+            await ctx.send(f"Encountered the following errors: {e}\n**Most likely need view audit log perms.**")
+            return
+
+        logger.info(f"[Ban initban()] retrieved audit log data for ban actions: {audit_logs}")
+        logger.info(f"[Ban initban()] retrieved ban list from guild: {bans}")
+
+        logger.info("[Ban initban()] Starting process to move all guild bans into db")
+        ban_ids = [ban.user.id for ban in bans]
+        ban_data = []
+        audit_users = []
+        for ban in audit_logs:
+            audit_users.append(ban.target.id)
+
+            ban_data.append([ban.target.name+'#'+ban.target.discriminator,
+                             ban.target.id,
+                             ban.user.name+'#'+ban.user.discriminator,
+                             ban.user.id,
+                             pytz.utc.localize(ban.created_at),
+                             ban.reason if ban.reason else 'No Reason Given!'
+                             ])
+
+        for ban in bans:
+            if ban.user.id in audit_users:
+                continue
+            ban_data.append([ban.user.name+'#'+ban.user.discriminator,
+                             ban.user.id,
+                             None,
+                             None,
+                             None,
+                             'No Reason Given!'
+                             ])
+
+        # push to db and ban_list
+        for ban in ban_data:
+            if ban[1] in ban_ids:
+                self.ban_list.append(ban[1])
+                await BannedUsers.insert_ban(ban[0], ban[1])
+
+            await BanRecords.insert_record(ban[0], ban[1], ban[2], ban[3], ban[4], ban[5])
+        await ctx.send(f"{len(ban_data)} moved from guild bans to db.")
+        logger.info(f"[Ban initban()] total of {len(ban_data)} bans moved into db")
```

Comment:
am I missing something or did you forget to remove the guilds bans when you made them wall_e bans?

Reply:
thats on purpose. incase something fucks up during the "conversion" process. once the command finishes mods can check the ban list, `.bans`, its looks right then they'll hit the ban purge button on the guild settings.